flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/AvgAggFunction.scala

```diff
@@ -137,6 +146,15 @@ abstract class BigIntegralAvgAggFunction[T] extends AggregateFunction[T] {
     }
   }
 
+  override def retract(accumulator: Accumulator, value: Any): Unit = {
```

Comment:
I assume you are suggesting implement LongAvgAggFunction directly with JTuple2[BigInteger, Long] as the accumulator, then yes, we could do that.